recipes-openxt/xenclient/ovmf/files/ovmf-q35-xen-support.patch

```diff
@@ -0,0 +1,662 @@
+   PCI_IO_DEVICE     *RootBridge;
+   EFI_HANDLE        ThisHostBridge;
+   LIST_ENTRY        *CurrentLink;
+@@ -827,8 +828,30 @@ StartPciDevices (
+ 
+     CurrentLink = CurrentLink->ForwardLink;
+   }
++#else
++  PCI_IO_DEVICE     *RootBridge;
++  LIST_ENTRY        *CurrentLink;
++
++  CurrentLink = mPciDevicePool.ForwardLink;
++
++  while (CurrentLink != NULL && CurrentLink != &mPciDevicePool) {
++
++    RootBridge = PCI_IO_DEVICE_FROM_LINK (CurrentLink);
++    //
++    // Locate the right root bridge to start
++    //
++      StartPciDevicesOnBridge (
++         RootBridge->Handle,
++         RootBridge,
++         NULL,
++         NULL,
++         NULL
++         );
+ 
++    CurrentLink = CurrentLink->ForwardLink;
++  }
+   return EFI_SUCCESS;
++#endif
+ }
+ 
+ /**
+diff --git a/MdeModulePkg/Bus/Pci/PciHostBridgeDxe/PciRootBridgeIo.c b/MdeModulePkg/Bus/Pci/PciHostBridgeDxe/PciRootBridgeIo.c
+index cda9b49..e12a51b 100644
+--- a/MdeModulePkg/Bus/Pci/PciHostBridgeDxe/PciRootBridgeIo.c
++++ b/MdeModulePkg/Bus/Pci/PciHostBridgeDxe/PciRootBridgeIo.c
+@@ -98,24 +98,28 @@ CreateRootBridge (
+   if (Bridge->Mem.Base < Bridge->Mem.Limit) {
+     ASSERT (Bridge->Mem.Limit < SIZE_4GB);
+     if (Bridge->Mem.Limit >= SIZE_4GB) {
++      ASSERT(!"Root bridge sadness #0");
+       return NULL;
+     }
+   }
+   if (Bridge->MemAbove4G.Base < Bridge->MemAbove4G.Limit) {
+     ASSERT (Bridge->MemAbove4G.Base >= SIZE_4GB);
+     if (Bridge->MemAbove4G.Base < SIZE_4GB) {
++      ASSERT(!"Root bridge sadness #1");
+       return NULL;
+     }
+   }
+   if (Bridge->PMem.Base < Bridge->PMem.Limit) {
+     ASSERT (Bridge->PMem.Limit < SIZE_4GB);
+     if (Bridge->PMem.Limit >= SIZE_4GB) {
++      ASSERT(!"Root bridge sadness #2");
+       return NULL;
+     }
+   }
+   if (Bridge->PMemAbove4G.Base < Bridge->PMemAbove4G.Limit) {
+     ASSERT (Bridge->PMemAbove4G.Base >= SIZE_4GB);
+     if (Bridge->PMemAbove4G.Base < SIZE_4GB) {
++      ASSERT(!"Root bridge sadness #3");
+       return NULL;
+     }
+   }
+@@ -131,6 +135,7 @@ CreateRootBridge (
+     if ((Bridge->PMem.Base < Bridge->PMem.Limit) ||
+         (Bridge->PMemAbove4G.Base < Bridge->PMemAbove4G.Limit)
+         ) {
++      ASSERT(!"Root bridge sadness #4");
+       return NULL;
+     }
+   }
+@@ -145,9 +150,11 @@ CreateRootBridge (
+     if ((Bridge->MemAbove4G.Base < Bridge->MemAbove4G.Limit) ||
+         (Bridge->PMemAbove4G.Base < Bridge->PMemAbove4G.Limit)
+         ) {
++      ASSERT(!"Root bridge sadness #5");
+       return NULL;
+     }
+   }
++  DEBUG ((EFI_D_INFO, "FISH FISH FISH Rootbridge OK\n"));
+ 
+   RootBridge = AllocateZeroPool (sizeof (PCI_ROOT_BRIDGE_INSTANCE));
+   ASSERT (RootBridge != NULL);
+diff --git a/OvmfPkg/AcpiPlatformDxe/AcpiPlatform.c b/OvmfPkg/AcpiPlatformDxe/AcpiPlatform.c
+index 61166c6..8840084 100644
+--- a/OvmfPkg/AcpiPlatformDxe/AcpiPlatform.c
++++ b/OvmfPkg/AcpiPlatformDxe/AcpiPlatform.c
+@@ -14,6 +14,12 @@
+ 
+ #include "AcpiPlatform.h"
+ 
++#define FISH \

```

Comment:
I get this for development, but maybe a more meaningful debug augmentation would be better.